### PR TITLE
Interface listing is not working on Android

### LINKED ISSUE #78: [UTIL] Interface listing is not working on Android
The dependency nix on its version 0.8 does not compile on Android due to a bug in it. Updating it to 0.22 and making a few adjustments on the code makes the compilation ok again, but the final application does not linking due to missing `getifaddrs` on Android.

May I suggest we delegate listing interfaces to a third party crate? https://crates.io/crates/get_if_addrs/0.5.3

Otherwise, we have to compile the C code that implements getifaddrs:
https://github.com/maidsafe-archive/get_if_addrs/blob/master/get_if_addrs-sys/build.rs

Once the appropriate approach is decided. I will gladly open a Pull Request

### PR DESCRIPTION
**Pull request by Andrepuel**
Fixes https://github.com/webrtc-rs/webrtc/issues/78

### CODE REVIEW

File: ifaddrs-android/LocalArray.h [hunk 1]
```diff
@@ -0,0 +1,66 @@
+/*
```
Comment (rainliu) [inline ifaddrs-android/LocalArray.h]: move ifaddrs-android as src/ifaces/ffi/android

File: ifaddrs-android/ifaddrs-android.cpp [hunk 2]
```diff
@@ -0,0 +1,199 @@
+/*
```
Comment (rainliu) [inline ifaddrs-android/ifaddrs-android.cpp]: rename ifaddrs-android.cpp to ifaddrs.cpp

**Review by rainliu — COMMENTED** [review]
please run cargo clippy and fmt to pass Github Actions

**Comment by Andrepuel** [comment]
All done!

**Pull request merged**